File: excel-blazor-add-in/Pages/Index.razor.js
/* Copyright(c) Maarten van Stam. All rights reserved. Licensed under the MIT License. */
/**
 * Basic function to show how to insert a value into cell A1 on the selected Excel worksheet.
 */
//office.ready(function () {    
  //export function helloButton() {

  //      return Excel.run(context => {

  //          // Insert text 'Hello world!' into cell A1.
  //          context.workbook.worksheets.getActiveWorksheet().getRange("A1").values = [['Hello world!']];

  //          // sync the context to run the previous API call, and return.
           
  //          return context.sync(); 
  //      });
    
  //  }
(window).lancerchaîné = async (event) => {
    console.log("lancerchaîné");
    try {
        callStaticLocalComponentMethod();
    }
    catch (err) {
        console.log();
        console.log("erreur lancer : " + err.message);
    }
    console.log("eventlancer Completed");
    event.completed();
}
(window).initialiser = async (event) => {
    console.log("initialiser");
    try {
        callStaticLocalComponentMethodinit();
    }
    catch (err) {
        console.log();
        console.log("erreur init : " + err.message);
    }
    console.log("eventinit  Completed");
    event.completed();
}
async function callStaticLocalComponentMethod() {
    console.log("avant");
    try {      
        var dato = "lancé";      
        dato = await DotNet.invokeMethodAsync("BlazorAddIn", "Localfunction");
        console.log("fin lancé : " + dato);       
    }
    catch (err) {
        console.log();
        console.log("erreur call : " + err.message);
    }
    finally {      
        console.log("aprèslancé : " + int2);
    }

}
async function callStaticLocalComponentMethodinit() {
    console.log("avant");
    try {       
        var dato = "init";       
        dato = await DotNet.invokeMethodAsync("BlazorAddIn", "Localfunctioninit");
        console.log("fin init : " + dato)       
    }
    catch (err) {
        console.log();
        console.log("erreur call : " + err.message);
    }
    finally {
        console.log("aprèsinit : " + int2);
    }
   
}

File: excel-blazor-add-in/Pages/Index.razor.cs
/* Copyright(c) Maarten van Stam. All rights reserved. Licensed under the MIT License. */

using Microsoft.AspNetCore.Components;
using Microsoft.JSInterop;

namespace BlazorAddIn.Pages
{
    /// <summary>
    /// Starter class to demo how to insert `Hello world!` text.
    /// </summary>
    
    public partial class Index
    {
        [Inject]
        public IJSRuntime JSRuntime { get; set; } = default!;
        public IJSObjectReference JSModule { get; set; } = default!;

        protected async override Task OnAfterRenderAsync(bool firstRender)
        {
            if (firstRender)
            {
                JSModule = await JSRuntime.InvokeAsync<IJSObjectReference>("import", "./Pages/Index.razor.js");               
            }
        }      
        /// Basic function to invoke inserting `Hello world!` text.
        /// </summary>
        //[JSInvokable]
        //private async Task HelloButton2() =>
        //    await JSModule.InvokeVoidAsync("helloButton");

        //[JSInvokable]
        //public static string LocalStaticMethod() => $"{nameof(LocalStaticMethod)} called";
        [JSInvokable]
        public static Task<int[]> Localfunction()
        {           
            return Task.FromResult(new int[] { 1, 2, 3 });
        }
       

    }
}
